File: server/app/schemas/conversation.py
from pydantic import BaseModel, Field
from datetime import datetime


class AskRequest(BaseModel):
    question: str = Field(min_length=1)
    top_k: int = Field(default=5, ge=1, le=10)


class SourceResponse(BaseModel):
    id: int
    document_id: int
    chunk_index: int
    text: str


class AskResponse(BaseModel):
    question: str
    answer: str
    sources: list[SourceResponse]


class ConversationCreateResponse(BaseModel):

    id: int


class MessageResponse(BaseModel):
    id: int
    role: str
    content: str


class ConversationResponse(BaseModel):
    id: int
    messages: list[MessageResponse]


class ConversationListItem(BaseModel):
    id: int
    created_at: datetime
    title: str | None


class ConversationListResponse(BaseModel):
    conversations: list[ConversationListItem]

File: client/lib/types/conversation.ts
export interface Message {
  id: number;
  role: "user" | "assistant";
  content: string;
}

export interface Conversation {
  id: number;
  messages: Message[];
}
export interface ConversationCreateResponse {
  id: number;
}

export interface ConversationListItem {
  id: number;
  created_at: string;
  title: string | null;
}

export interface ConversationListResponse {
  conversations: ConversationListItem[];
}

export interface AskRequest {
  question: string;
  top_k?: number;
}

export interface Source {
  id: number;
  document_id: number;
  chunk_index: number;
  text: string;
}

export interface AskResponse {
  question: string;
  answer: string;
  sources: Source[];
}
